src/plugins/data_source/server/saved_objects/credential_saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,278 @@
+      options?: SavedObjectsBulkUpdateOptions
+    ): Promise<SavedObjectsBulkUpdateResponse<T>> => {
+      objects = await Promise.all(
+        objects.map(async (object) => {
+          const { type, attributes } = object;
+
+          this.validateType(type);
+          this.validateAttributes(attributes);
+
+          const { title, authType, credentialMaterials, description } = attributes;
+
+          if (this._noAuthType === credentialMaterials.credentialMaterialsType) {
+            return object;
+          }
+
+          const encryptedAttributes = {
+            title,
+            authType,
+            credentialMaterials: await this.validateAndEncryptCredentialMaterials(
+              credentialMaterials
+            ),
+            description,
+          };
+
+          return ({
+            ...object,
+            attributes: encryptedAttributes,
+            // Unfortunately this throws a typescript error without the casting.  I think it's due to the
+            // convoluted way SavedObjects are created.
+          } as unknown) as SavedObjectsBulkUpdateObject<T>;
+        })
+      );
+
+      return await wrapperOptions.client.bulkUpdate(objects, options);
+    };
+
+    return {
+      ...wrapperOptions.client,
+      create: createWithCredentialMaterialsEncryption,
+      bulkCreate: bulkCreateWithCredentialMaterialsEncryption,
+      checkConflicts: wrapperOptions.client.checkConflicts,
+      delete: wrapperOptions.client.delete,
+      find: wrapperOptions.client.find,
+      bulkGet: wrapperOptions.client.bulkGet,
+      get: wrapperOptions.client.get,
+      update: updateWithCredentialMaterialsEncryption,
+      bulkUpdate: bulkUpdateWithCredentialMaterialsEncryption,
+      errors: wrapperOptions.client.errors,
+      addToNamespaces: wrapperOptions.client.addToNamespaces,
+      deleteFromNamespaces: wrapperOptions.client.deleteFromNamespaces,
+    };
+  };
+
+  private validateType(type: string) {
+    if (this._type !== type) {
+      throw SavedObjectsErrorHelpers.createUnsupportedTypeError(type);
+    }
+    return;
+  }
+
+  private validateAttributes<T = unknown>(attributes: T) {
+    const { title, authType, credentialMaterials } = attributes;
+
+    if (title === undefined) {
+      throw SavedObjectsErrorHelpers.createBadRequestError('attribute "title" required');
+    }
+
+    if (authType === undefined) {
+      throw SavedObjectsErrorHelpers.createBadRequestError('attribute "authType" required');
+    }
+
+    if (credentialMaterials === undefined) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        'attribute "credentialMaterials" required'
+      );
+    }
+
+    // TODO: Support other auth types https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2110
+    if (this._authType !== authType) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        `Unsupported authType type: '${attributes.authType}'`
+      );
+    }
+    return;
+  }
```

Comment:
same as above, this code cause errors to all other saved object types

Reply:
resolved